tests/openvpn/test_backend.py

```diff
@@ -882,3 +882,31 @@ def test_tls_auth_key_present(self):
         )
         client = OpenVpn(client_config)
         self.assertEqual(client.render(), self._openvpn_client_tls_auth_render)
+
+    def test_ca_same_file_path_for_same_device(self):
+        conf = {
+            "cert": "cert.pem",
+            "dev": "tap0",
+            "dev_type": "tap",
+            "dh": "dh.pem",
+            "key": "key.pem",
+            "mode": "server",
+            "name": "test-1",
+            "proto": "udp",
+            "tls_server": True,
+        }
+        template1 = {
+            "openvpn": [{"ca": "TEST", **conf}],
+            "files": [{"path": "/file-path/", "mode": "0644", "contents": "testing!"}],
+        }
+        template2 = {
+            "openvpn": [{"ca": "TEST", **conf}],
+            "files": [{"path": "/file-path/", "mode": "0644", "contents": "testing!"}],
+        }
+        o_t1 = OpenVpn(
+            {
+                "openvpn": [conf],
+            },
+            templates=[template1, template2],
+        )
+        self.assertEqual(o_t1.render(), o_t1.render())
```

Comment:
This assertion doesn't make sense.

I would first render the configuration in a try/except block and fail the test if the validation error is raised.
Then once the conf is correctly generated we can ensure there's only one occurrence of the ca file.